components/canvas/utils/FormatNodeText.tsx

```diff
@@ -0,0 +1,34 @@
+import { Typography, TypographyProps } from "@equinor/eds-core-react";
+import styles from "../Node.module.scss";
+
+export function FormatNodeText({
+  children,
+  ...typographyProps
+}: TypographyProps) {
+  if (!children) return null;
+  const text = children.toString();
+  // Pattern matching for Markdown hyperlink urls
+  const pattern = /\[([^\]]+)]\(([^)]+)\)/g;
+  const result = [];
+  let lastIndex = 0;
+
+  text.replace(pattern, (match, linkText, _, offset) => {
+    if (lastIndex <= offset) {
+      result.push(<>{text.slice(lastIndex, offset)}</>);
+    }
+
+    result.push(
+      <Typography className={styles[typographyProps.className ?? ""]} link>
```

Comment:
We should check that className is not undefined to avoid the strict mode error

Reply:
fixed